compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -109,12 +109,25 @@ object RefChecks {
       for (reqd <- cinfo.cls.givenSelfType.classSymbols)
         checkSelfConforms(reqd, "missing requirement", "required")
 
+      def illegalEnumFlags = !cls.isOneOf(Enum | Trait)
```

Comment:
For boolean predicates, it's a good idea to start them with is/has:
```suggestion
      def hasIllegalEnumFlags = !cls.isOneOf(Enum | Trait)
```